mistralrs-core/src/pipeline/mod.rs

```diff
@@ -457,10 +396,13 @@ pub struct GeneralMetadata {
     pub repeat_last_n: usize,
     pub tok_trie: Arc<TokTrie>,
     pub has_no_kv_cache: bool,
-    pub is_xlora: bool,
     pub num_hidden_layers: usize,
     pub eos_tok: Vec<u32>,
-    pub is_lora: bool,
+    // Model utilizes an adapter (LoRA or X-LoRA):
+    pub has_adapter: bool,
+    // TODO: There is a possibility that code mistakenly queries only `is_xlora`,
+    // rather than `has_adapter`. Thus may actually only support X-LoRA? (potential bugs?)
+    pub is_xlora: bool,
```

Comment:
Might need your input here for clarification if `is_xlora` is exclusive to X-LoRA or also represents plain LoRA, and if `AdapterKind` were to have another variant is there value in separate `has_adapter` and `is_xlora`?

The enum itself `AdapterKind` could potentially be stored instead in future to cover both needs.